boa/profiling.py

```diff
@@ -102,34 +117,267 @@ def merge(self, other: "LineProfile") -> None:
+            call_profile["median"] = int(statistics.median(gas_used))
+            call_profile["min"] = min(gas_used)
+            call_profile["max"] = max(gas_used)
+            if len(gas_used) == 1:
+                call_profile["stdev"] = 0
+            else:
+                call_profile["stdev"] = int(statistics.stdev(gas_used))
+            call_profiles[key] = call_profile
+            average_gas_costs.append(call_profile["mean"])
+
+        contract_profiles[contract] = call_profiles
+        max_avg_gas.append(max(average_gas_costs))
+
+    # arrange from most to least expensive contracts:
+    sort_gas = sorted(zip(max_avg_gas, profiled_contracts), reverse=True)
+    sorted_contracts = [x for _, x in sort_gas]
+
+    for contract in sorted_contracts:
+        call_profiles = contract_profiles[contract]
+        c = 0
+        for key in call_profiles.keys():
+            profile = call_profiles[key]
+            cname = ""
+            caddr = ""
+            if c == 0:
+                cname = profile["contract_name"]
+                caddr = contract
+            table.add_row(
+                cname,
+                caddr,
+                key,
+                str(profile["count"]),
+                str(profile["mean"]),
+                str(profile["median"]),
+                str(profile["stdev"]),
+                str(profile["min"]),
+                str(profile["max"]),
+            )
+            c += 1
+
+        table.add_section()
+
+    return table
+
+
+def get_line_profile_table(env: Env):
+
+    contracts = {}
+    for lp, gas_data in env._cached_line_profiles.items():
+        contract_uid = (lp.contract_name, lp.address)
+
+        # add spaces so numbers take up equal space
+        lineno = str(lp.lineno)
+        if lp.lineno < 10:
+            lineno = "   " + lineno
+        if 10 <= lp.lineno < 100:
+            lineno = "  " + lineno
+        if 100 <= lp.lineno < 1000:
+            lineno = " " + lineno
+
+        gas_data = (lineno + ": " + lp.line_src, gas_data)
+        if contract_uid not in contracts:
+            contracts[contract_uid] = [gas_data]
+        else:
+            contracts[contract_uid].append(gas_data)
+
+    table = _create_table(show_contract=False)
+    for (contract_name, _), _data in contracts.items():
+
+        l_profile = []
+        for code, gas_used in _data:
+
+            if len(gas_used) == 1:
+                stdev = 0
+            else:
+                stdev = int(statistics.stdev(gas_used))
+
+            if code.endswith("\n"):
+                code = code[:-1]
+
+            l_profile.append(
+                [
+                    contract_name,
+                    code,
+                    len(gas_used),
+                    int(statistics.mean(gas_used)),
+                    int(statistics.median(gas_used)),
+                    stdev,
+                    min(gas_used),
+                    max(gas_used),
+                ]
```

Comment:
style: unless the item should be mutable, make it a tuple

Reply:
since this code is similar to the long block in `get_call_profile_table()`, can we factor it into a Stats object?